clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -880,4 +880,25 @@ private HeartbeatRequestManager createHeartbeatRequestManager(
                 backgroundEventHandler,
                 metrics);
     }
+
+
+    @Test
+    public void reproduceHeartbeatFail() throws InterruptedException {
```

Comment:
I'm afraid in some time from now we won't know what heartbeat failure this test refers to (without reading the whole test). So what about renaming this to something like `testLeaveGroupSentOnCloseWhenPreviousOneInflight` or something along those lines